app.rb

```diff
@@ -0,0 +1,199 @@
+    case entry
+    when 1
+      create_student
+    when 2
+      create_teacher
+    end
+  end
+
+  def all_people
+    key = 1
+    puts
+    puts 'People'.upcase
+    puts
+    puts 'No people yet! Choose option 3 to add a person ' if @people.empty?
+    @people.each do |person|
+      print "#{key} - [#{person.class.name} ID]: #{person.id} Age: #{person.age} Name: #{person.name} "
+      print "Parent Permission: #{person.parent_permission}" if person.is_a?(Student)
+      print "Specialization: #{person.specialization}" if person.is_a?(Teacher)
+      puts
+      key += 1
+    end
+  end
+
+  def all_books
+    key = 1
+    puts
+    puts 'Books'.upcase
+    puts
+    puts 'No book yet! Choose option 4 to add a book ' if @books.empty?
+    @books.each do |book|
+      puts "#{key} - #{book.title} by #{book.author}"
+      key += 1
+    end
+  end
+
+  def create_teacher
+    age = -1
+    print 'Enter Age: '
+    while age <= 0
+      age = gets.chomp.to_i
+      print 'Enter valid age for Teacher: ' if age <= 0
+    end
+
+    print 'Enter Name: '
+    name = gets.chomp.strip.capitalize
+
+    print 'Enter Specialization: '
+    specialization = gets.chomp.strip.capitalize
+
+    @people << Teacher.new(age, name, specialization)
+    puts ' Teacher was created successfuly! '
+  end
+
+  def create_student
+    print 'Enter Age: '
+    print 'Age: '
+    age = gets.chomp
+
+    print 'Enter Name: '
+    name = gets.chomp.strip.capitalize
+
+    print 'Does student have parent permission? (Y/N): '
+    permission = gets.chomp.strip.upcase
+
+    case permission
+    when 'Y', 'YES'
+      permission = true
+    when 'N', 'NO'
+      permission = false
+    end
+    @people << Student.new(age, name, parent_permission: permission)
+    puts
+    puts 'Student was created successfuly!'
+    puts
+  end
+
+  def create_book
+    print 'Enter Title: '
+    title = gets.chomp.strip.capitalize
+
+    print 'Enter Author: '
+    author = gets.chomp.strip.capitalize
+
+    @books << Book.new(title, author)
+    puts
+    puts 'Book was created successfully!'
+    puts
+  end
+
+  def create_rental
+    all_books
+    print 'Select the key of the book: '
+    book_select = gets.chomp.chomp.to_i
+    all_people
+    print 'Select the key of the person: '
+    person_select = gets.chomp.chomp.to_i
+
+    print 'Select the date: (Year/Month/Day): '
+    date = gets.chomp.strip
+    book = @books[book_select]
+    person = @people[person_select]
+    new_rental = Rental.new(date, book, person)
+    @all_rentals.push(new_rental)
+    puts
+    puts 'Rental was created successfuly!'
+    puts
+  end
+
+  def list_rentals
+    puts
+    all_people
+    puts 'Choose person ID: '
+    entry = gets.chomp.to_i
+
+    puts 'Rentals'.upcase
+    puts
+    @people.each do |person|
+      next unless person.id == entry
+
+      @all_rentals.each do |rental|
+        puts "Rental date: #{rental.date} - #{rental.book} by #{rental.person}"
+      end
+    end
+  end
```

Comment:
There is an issue with creating the rentals. Kindly refer to [this video](https://www.loom.com/share/fcdb085d2edd4c42b707b97932d19ee9) where I talked in detail about the problem. The problem was too complex, hence I created a video to try to simplify it.